front/src/Phaser/Game/GameMapPropertiesListener.ts

```diff
@@ -36,42 +52,178 @@ export class GameMapPropertiesListener {
-                            allProps.get("openWebsiteAllowApi") as boolean | undefined,
-                            allProps.get("openWebsitePolicy") as string | undefined,
-                            allProps.get("openWebsiteWidth") as number | undefined
-                        );
+                            allowApiProperty,
+                            websitePolicyProperty,
+                            websiteWidthProperty,
+                            websitePositionProperty,
+                        ).then(coWebsite => {
+                            const coWebsiteOpen = this.coWebsitesOpenByLayer.get(layer);
+                            if (coWebsiteOpen && coWebsiteOpen.state === OpenCoWebsiteState.MUST_BE_CLOSE) {
+                                coWebsiteManager.closeCoWebsite(coWebsite);
+                                this.coWebsitesOpenByLayer.delete(layer);
+                                this.coWebsitesActionTriggerByLayer.delete(layer);
+                            } else {
+                                this.coWebsitesOpenByLayer.set(layer, {
+                                    coWebsite,
+                                    state: OpenCoWebsiteState.OPENED
+                                });
+                            }
+                        });
 
-                        layoutManagerActionStore.removeAction("openWebsite");
+                        layoutManagerActionStore.removeAction(actionUuid);
                     };
-                    const openWebsiteTriggerValue = allProps.get(TRIGGER_WEBSITE_PROPERTIES);
-                    if (openWebsiteTriggerValue && openWebsiteTriggerValue === ON_ACTION_TRIGGER_BUTTON) {
-                        let message = allProps.get(WEBSITE_MESSAGE_PROPERTIES);
-                        if (message === undefined) {
-                            message = "Press SPACE or touch here to open web site";
+
+                    if (websiteTriggerProperty && websiteTriggerProperty === ON_ACTION_TRIGGER_BUTTON) {
+                        if (!websiteTriggerMessageProperty) {
+                            websiteTriggerMessageProperty = "Press SPACE or touch here to open web site";
                         }
+
+                        this.coWebsitesActionTriggerByLayer.set(layer, actionUuid);
+
                         layoutManagerActionStore.addAction({
-                            uuid: "openWebsite",
+                            uuid: actionUuid,
                             type: "message",
-                            message: message,
+                            message: websiteTriggerMessageProperty,
                             callback: () => openWebsiteFunction(),
                             userInputManager: this.scene.userInputManager,
                         });
                     } else {
                         openWebsiteFunction();
                     }
-                }
+                });
+            };
+
+            handler();
```

Comment:
Why do we need to set the code in a "handler" function that is called just after being defined?